Automattic/wp-calypso
# Add HSTS Notice for Required TLD's

#### Overview

Display an information icon with a tooltip to inform users when a suggested domain requires HSTS. The tooltip will reveal more information on this. See https://github.com/Automattic/wp-calypso/issues/40973

Before:
![Screenshot on 2020-09-09 at 09-40-11](https://user-images.githubusercontent.com/8639742/92591385-b2152300-f295-11ea-867e-4db464003d02.png)

After:
![Screenshot on 2020-09-09 at 09-39-08](https://user-images.githubusercontent.com/8639742/92591403-b8a39a80-f295-11ea-9c30-3436dd82a006.png)

#### Changes proposed in this Pull Request

* Created a new `InfoTooltip` component within `packages/domain-picker`, with a view for this to be replaced by the `InfoPopover` component once it gets moved from `client` to `@automattic/components` (See https://github.com/Automattic/wp-calypso/issues/45449)
* Added the `hsts_required` property to the `domain-suggestion` types, this is returned with the suggested domains to indicate whether they require HSTS.
* Changes are accompanied by component tests.
* Existing component tests for `domain-picker` had to be skipped due to them becoming stale as they were never getting run. See issue raised to fix this here https://github.com/Automattic/wp-calypso/issues/45501

#### Testing instructions

* Create a new site at [/new](https://hash-64a044e58c55f0d40f75328c7271ffbcfa456b52.calypso.live/new), include a domain in its name that will require HSTS (such as .dev). This will ensure its brought back as a suggestion.
* The .dev suggestion of your domain should have an information icon next to it.
* Clicking this icon should reveal a popover explaining HSTS is required and how they can learn more
* Domains that do not require HSTS should not show an information icon next to it.

**Automated tests**
* Also running `npm run test-packages` will run the component tests.

Fixes https://github.com/Automattic/wp-calypso/issues/40973
